RELEASE_NOTES.md

```diff
@@ -1,3 +1,112 @@
+## Cookstyle 5.2
+
+### New Chef Cops
+
+#### Chef/InvalidPlatformMetadata
+
+The `InvalidPlatformMetadata` cop detects invalid platform names being used in the supports metadata.rb method. It uses a hardcoded set of common typos so it's not able to detect all invalid platforms, but it does detect the most common mistakes found on the Supermarket.
+
+`Enabled by default`: True
+
+`Autocorrects`: True (deletes the occurrence)
+
+#### Chef/RespondToResourceName
+
+The `RespondToResourceName` cop detects the usage of respond_to?(:resource_name) in resources to provide backwards compatibility. This `respond_to?` check is no longer necessary in Chef Infra Client 12.5+
+
+`Enabled by default`: True
+
+`Autocorrects`: True (deletes the `if respond_to?` occurence)
+
+#### Chef/RespondToProvides
+
+The `RespondToProvides` cop detects the usage of respond_to?(:provides) in providers to provide backwards compatibility. This `respond_to?` check is no longer necessary in Chef Infra Client 12+
+
+`Enabled by default`: True
+
+`Autocorrects`: True (deletes the `if respond_to?` occurence)
+
+#### Chef/RespondToInMetadata
+
+The `RespondToInMetadata` cop detects the usage of `if respond_to?(:foo)` to gate newer metadata methods from running on older Chef Infra Client releases. In Chef Infra Client 12.15 and later this is no longer necessary as chef-client will no longer fail if unknown metadata is encoutered.
+
+`Enabled by default`: True
+
+`Autocorrects`: True (deletes the `if respond_to?` occurence)
+
+#### Chef/LongDescriptionMetadata
+
+The `LongDescriptionMetadata` cop detects usage of `long_description` in metadata.rb. The `long_description` metadata was never utilized by Chef Infra or Supermarket, but cookbooks commonly load their README.md file into the metadata. This increases the size of metadata stored on Chef Infra Server and provides no benefit to users.
+
+`Enabled by default`: True
+
+`Autocorrects`: True (deletes the `if respond_to?` occurence)
+
+#### Chef/CookbooksDependsOnSelf
+
+The `CookbooksDependsOnSelf` cop detects a cookbook that depends on itself in metadata.rb.
+
+`Enabled by default`: True
+
+`Autocorrects`: False
+
+#### Chef/MetadataMissingName
+
+The `MetadataMissingName` cop detects metadata.rb which is missing the name field required by Chef Infra Client 12+
+
+`Enabled by default`: True
+
+`Autocorrects`: False
+
+#### Chef/RequireRecipe
+
+The `RequireRecipe` cop detects the usage of `require_recipe` to include a recipe instead of `include_recipe`.
```

Comment:
Is this required in all versions of Chef Infra Client, or just in more recent versions?

Reply:
for the last ~8 years I think. Basically forever